src/operator.py

```diff
@@ -0,0 +1,339 @@
+    print("Upload_mission service finished!")
+    return True
+
+
+@custom_service
+def set_speed(request, response):
+    print("Calling set_speed service...")
+    if not drones[request["drone_id"]]:
+        response["success"] = False
+        response["message"] = "No drone with that id."
+        response["drone_id"] = -1
+        return False
+    print('Setting speed to {}...'.format(request['data']))
+    response = drones[request["drone_id"]].set_speed(request["data"])
+    response['success'] = True
+    print("Set_speed service finished!")
+    return True
+
+
+# includes startmission, pausemission, resume mission, landdrone, flyhome
+@custom_service
+def control_drone(request, response):
+    print("Calling control_drone service...")
+    control_task = request["control_task"]
+    drone = drones.get(request["id"])
+    if control_task == "start_mission":
+        print("Executing start_mission...")
+        response = drone.start_mission()
+    elif control_task == "pause_mission":
+        print("Executing pause_mission...")
+        response = drone.pause_mission()
+    elif control_task == "resume_mission":
+        print("Executing resume_mission...")
+        response = drone.resume_mission()
+    elif control_task == "land_drone":
+        print("Executing land_drone...")
+        response = drone.land_drone()
+    elif control_task == "fly_home":
+        print("Executing fly_home...")
+        response = drone.fly_home()
+    else:
+        response["success"] = False
+        response["message"] = "Invalid control task"
+        return False
+    print("Control_drone service finished!")
+    return True
+
+
+@custom_service
+def query_topics(request, response):
+    print("Calling query_topics service...")
+    id = request["id"]
+    response["id"] = id
+    all_topics_response = []
+    if id == 0:
+        for k,v in all_topics.items():
+            all_topics_response.append({"name": k, "type": v})
+    else:
+        if id in drones:
+            all_topics_response = drones[id].topics
+            for sensor_id in drones[id].sensors:
+                all_topics_response += sensors[sensor_id].topics
+        else if id in sensors:
+            all_topics_response = sensors[id].topics
+        else:
+            response["success"] = False
+            response["message"] = "No drone or sensor with that id."
+            return False
+
+    response["all_topics"] = all_topics_response
+    response["success"] = True
+    response["message"] = "Successfully queried topics."
+    print(all_topics_response)
+    print("Query_topics service finished!")
+    return True
+
+############################
+# Drone -> Server Handlers #
+############################
+@custom_service
+def register_drone(request, response):
+    '''
+    :param request: dict of {drone_name: string, drone_type: string}
+    '''
+    print("Calling register_drone service...")
+    def get_id():
+        global next_id
```

Comment:
For now I made it so that Drone ids are odd and Sensor ids are even. I didn't do the string prefix option for now, just because I don't know what I will have to change yet.